_posts/2018-05-07-A-Machine-Learning-Example-For-Business.md

```diff
@@ -0,0 +1,271 @@
+
+There are dozens of predictive machine learning algorithms that work in a different way, but a typical situation is to create a table like this, where the last column is the variable that you want to predict. For historical data the predicted variable is known already. All or part of the other variables are predictor variables.
+
+![Result variable in sample data](/img/a-machine-learning-example-for-business/sample-data-result-variable.png)
+
+## Predicting from reports is not systematic
+In traditional reporting you could analyze the rate of late arriving orders from previously introduced 20 rows of data by doing basic data aggregation and see how many percent of orders have arrived late on average for each variable combination.
+
+![Summary report](/img/a-machine-learning-example-for-business/summary-report.png)
+
+Using reports to forecast future is problematic, however. If there's only one record for a combination such as `Mon`, `L` and `Express` in row 1, the probability is always 0 or 1, which can't be true. The summary report also does not handle individual variables (columns) as independent entities, and that's why there is no way of evaluating probability for previously non existing combinations such as `Tue`, `M` and `Go`.
+
+And the most important part: There's no way of evaluating how these insights fit into new wholesale orders.
+
+## Concept of training and testing in machine learning
+The secret of predictive machine learning is to split the process in two phases:
+1. Training the model
+2. Testing the model
+
+The purpose is, that after these two stages you should be confident whether your <i>model</i> works for the particular case or not.
+
+For testing and training, the data has to be split in two parts: Training data and testing data. Let's use a rule of thumb and take 70% of rows for training and 30% for testing. Note, that all the data is historical and thus the actual outcome for all records are known.
+
+First 14 rows (70%) are for model training.
+![Training data](/img/a-machine-learning-example-for-business/machine-learning-example-training-data.png)
+
+In the <b>training phase</b> the selected algorithm produces a <i>model</i>. A model isn't very specific term, as it can be for example:
+* Another algorithm.
+* A formula such as <i>2x+y+3z-1</i>.
+* A lookup table.
+* Any other kind of data structure.
+* A decision tree.
+
+The important thing is, that the produced <i>model</i> should be something that you can use to make predictions in the <b>testing phase</b>. The situation is very fruitful, as now you can make a prediction for each record in testing data with the model, but you can also see, what was the actual outcome for that observation.
+
+The trained model will be tested with the last 6 rows (30%).
+![Testing data](/img/a-machine-learning-example-for-business/machine-learning-example-testing-data.png)
+
+## Selecting the machine learning algorithm
```

Comment:
Here, the motivation for selecting the algorithm is not discussed. In that sense, the subtitle can be thought to be a bit misleading.

Reply:
I justified the selection of this algorithm in more detail.